### PR TITLE
fix dns endpoint controller merged two different type record's targets with same dns name  

### PR DESCRIPTION
**Pull request by Dingshujie**
**What happened:**
we have cluster A in zone A, cluster B in zone A,
now we create a federated service nginx, deploy service only in cluster A, then create a servicednsrecord associate with this federated service,  and dns endpoint will generate two zone-level endpoint with same dns name.
clusterA: nginx.default.dsj-test.svc.cn-north-7b.cn-north-7.jessiading.com:Record A: xxx.xxx.xxx.xxx
clusterB: nginx.default.dsj-test.svc.cn-north-7b.cn-north-7.jessiading.com:Record CNAME: nginx.default.dsj-test.svc.cn-north-7.jessiading.com
finally, controller will merge two endpont into one,  nginx.default.dsj-test.svc.cn-north-7b.cn-north-7.jessiading.com:Record A: xxx.xxx.xxx.xxx,nginx.default.dsj-test.svc.cn-north-7.jessiading.com
 when external-dns create this endpoint in provider,  get error response, because Record A targets must be ipv4 address list.

### CODE REVIEW

File: pkg/controller/dnsendpoint/common.go [hunk 1]
```diff
@@ -118,8 +118,14 @@ func DedupeAndMergeEndpoints(endpoints []*feddnsv1a1.Endpoint) (result []*feddns
 	for i := 1; i < len(endpoints); {
 		if endpoints[i].DNSName == endpoints[i-1].DNSName {
 			// Merge targets
-			endpoints[i-1].Targets = append(endpoints[i-1].Targets, endpoints[i].Targets...)
-			endpoints[i-1].Targets = sortAndRemoveDuplicateTargets(endpoints[i-1].Targets)
+			if endpoints[i-1].RecordType != endpoints[i].RecordType {
+				if endpoints[i-1].RecordType == RecordTypeCNAME {
```
Comment (shashidharatd) [inline pkg/controller/dnsendpoint/common.go]: (Suggestion) Better to add a comment on why we are doing this, which will help future maintainers.

Comment (Dingshujie) [reply pkg/controller/dnsendpoint/common.go]: done~

Comment (shashidharatd) [reply pkg/controller/dnsendpoint/common.go]: s/enpoint/endpoint/
s/at same region or at same az/at same region and az/
s/if cluster no endpoints/if cluster does not have endpoints/

Comment (Dingshujie) [reply pkg/controller/dnsendpoint/common.go]: thanks. 

fixed. ^^

**Comment by shashidharatd** [comment]
/lgtm
/approve
Thanks @Dingshujie

**Pull request merged**